### PR TITLE
Add some examples for Task related functions and Examples cleanup

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/task.jl [hunk 1]
```diff
@@ -543,6 +556,24 @@ end
...
+julia> errormonitor(b);
+
+julia> schedule(b);
+
+Unhandled Task ERROR: task failed
+Stacktrace:
+ [1] error(s::String)
+   @ Base ./error.jl:35
+ [2] a4()
+   @ Main ./REPL[1]:1
```
Comment (vtjnash) [inline base/task.jl:575]: commonly, they are all stacked on the same line as:
```suggestion
julia> a4() = error("task failed");

julia> Base._wait(schedule(errormonitor(Task(a4))))
Unhandled Task ERROR: task failed
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] a4()
   @ Main ./REPL[1]:1
```

Comment (vtjnash) [reply base/task.jl:575]: Or perhaps more canonically using the spawn macro:
```suggestion
julia> Base._wait(errormonitor(Threads.@spawn error("task failed")))
Unhandled Task ERROR: task failed
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:35
 [2] (::var"#5#6")()
   @ Main ./task.jl:501
```

**Review by vtjnash — APPROVED** [review]

File: base/task.jl [hunk 2]
```diff
@@ -450,9 +450,22 @@ const sync_varname = gensym(:sync)
...
+a [`CompositeException`](@ref).
+
+# Examples
+```julia-repl
+julia> Threads.nthreads()
+4
+
+julia> @sync begin
+           Threads.@spawn println("Thread-id $(Threads.threadid()), task 1")
+           Threads.@spawn println("Thread-id $(Threads.threadid()), task 2")
```
Comment (fredrikekre) [inline base/task.jl:464]:
```suggestion
           Threads.@spawn println("Thread-id \$(Threads.threadid()), task 1")
           Threads.@spawn println("Thread-id \$(Threads.threadid()), task 2")
```

File: base/task.jl [hunk 3]
```diff
@@ -543,6 +556,17 @@ end
...
+
+# Examples
+```julia-repl
+julia> Base._wait(errormonitor(Threads.@spawn error("task failed")))
+Unhandled Task ERROR: task failed
+Stacktrace:
+ [1] error(s::String)
+   @ Base ./error.jl:35
+ [2] a4()
+   @ Main ./REPL[1]:1
```
Comment (fredrikekre) [inline base/task.jl:568]:
```suggestion
[...]
```

**Pull request merged**